File: java/src/main/java/gov/doc/ntia/sigmf/examples/algorithm/FrequencyDomainDetectionExample.java
package gov.doc.ntia.sigmf.examples.algorithm;

import gov.doc.ntia.sigmf.Annotation;
import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.annotation.algorithm.FrequencyDomainDetection;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class FrequencyDomainDetectionExample implements Example {
  @Override
  public MetaDoc getExample() {
    FrequencyDomainDetection fdd = new FrequencyDomainDetection();
    fdd.setNumberOfSamplesInFft(1024);
    fdd.setSampleStart(0L);
    fdd.setDetector("fft_mean_power");
    fdd.setSampleCount(1024L);
    fdd.setReference("antenna output");
    List<Annotation> annotations = new ArrayList();
    annotations.add(fdd);

    FrequencyDomainDetection fdd2 = new FrequencyDomainDetection();
    fdd2.setNumberOfSamplesInFft(1024);
    fdd2.setSampleStart(1024L);
    fdd2.setDetector("fft_max_power");
    fdd2.setSampleCount(1024L);
    fdd2.setReference("antenna output");
    annotations.add(fdd2);
    MetaDoc metaDoc = new MetaDoc();
    Global global = ExampleUtils.getGlobal();
    metaDoc.setGlobal(global);
    metaDoc.setAnnotations(annotations);
    return metaDoc;
  }
}

File: Csharp/sigmf/sigmf/examples/algorithm/FrequencyDomainDetectionExample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.algorithm
{
    public class FrequencyDomainDetectionExample : Example
    {
        public override MetaDoc getExample()
        {
            MetaDoc metaDoc = new MetaDoc();
            Globals globals = ExampleUtils.getGlobals(getCurrentTime());
            List<Annotation> annotations = new List<Annotation>();
            FrequencyDomainDetection fdd = new FrequencyDomainDetection();
            fdd.NumberOfSamplesInFft = 1024;
            fdd.SampleStart = 01;
            fdd.Detector = "fft_mean_power";
            fdd.DetectionDomain = "frequency";
            fdd.SampleCount = 10241;
            fdd.Reference = "antenna output";
            annotations.Add(fdd);

            FrequencyDomainDetection fdd2 = new FrequencyDomainDetection();
            fdd2.NumberOfSamplesInFft = 1024;
            fdd2.SampleStart = 10241;
            fdd2.Detector = "fft_max_power";
            fdd2.DetectionDomain = "frequency";
            fdd2.SampleCount = 10241;
            fdd2.Reference = "antenna output";
            annotations.Add(fdd2);
            metaDoc.setGlobal(globals);
            metaDoc.setAnnotations(annotations);
            return metaDoc;

        }
        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}
